bench/workgen/workgen.cxx

```diff
@@ -2701,6 +2888,75 @@ WorkloadRunner::run_all(WT_CONNECTION *conn)
         }
     }
 
+    // Start a thread to create tables
+    pthread_t tables_create_thandle;
+    WorkloadRunnerConnection *tableCreate = nullptr;
+    if (options->create_interval > 0) {
+        if (options->create_count < 1) {
+            std::cerr << "create_count needs to be greater than 0" << std::endl;
+            stopping = true;
+        } else if (options->create_target > 0 && options->create_trigger <= 0) {
+            std::cerr << "Need to specify a create_trigger when setting a create_target."
+                      << std::endl;
+            stopping = true;
+        } else if (options->create_trigger > 0 && options->create_target <= 0) {
+            std::cerr << "Need to specify a create_target when setting a create_trigger."
+                      << std::endl;
+            stopping = true;
+        } else if (options->create_trigger != 0 &&
+          options->create_target < options->create_trigger) {
+            std::cerr << "create_target should be greater than create_trigger." << std::endl;
+            stopping = true;
+        } else {
+            tableCreate = new WorkloadRunnerConnection();
+            tableCreate->runner = this;
+            tableCreate->connection = conn;
+
+            if ((ret = pthread_create(&tables_create_thandle, nullptr,
+                   thread_tables_create_workload, tableCreate)) != 0) {
+                std::cerr << "pthread_create failed err=" << ret << std::endl;
+                std::cerr << "Stopping table create thread." << std::endl;
+                (void)pthread_join(tables_create_thandle, &status);
```

Comment:
If the pthread_create failed, won't the handle be unusable (i.e. pthread_join may crash)?

Reply:
That is a good catch, this is copy paste error on my part as I copied the code for other threads. I remember thinking this is off and then paged out of my head. Thanks, I have fixed at other places as well.